src/encord_active/lib/common/filtering.py

```diff
@@ -33,15 +38,28 @@ def __hash__(self):
         return hash(json.dumps(self.dict(exclude_defaults=True)))
 
 
-def apply_filters(df: pd.DataFrame, filters: Filters):
+def apply_filters(df: pd.DataFrame, filters: Filters, pfs: ProjectFileStructure):
     filtered = df.copy()
     filtered["data_row_id"] = filtered.index.str.split("_", n=3).str[0:3].str.join("_")
     filtered["is_label_metric"] = filtered.index.str.split("_", n=3).str.len() > 3
 
+    # Add workflow stage column in workflow projects
+    if is_workflow_project(pfs):
+        filtered["label_hash"] = filtered.index.str.split("_", n=1).str[0]
+        lr_metadata = json.loads(pfs.label_row_meta.read_text(encoding="utf-8"))
+        lr_to_workflow_stage = {
+            lr_hash: metadata.get("workflow_graph_node", dict()).get("title", None)
+            for lr_hash, metadata in lr_metadata.items()
+        }
+        filtered["workflow_stage"] = filtered["label_hash"].map(lr_to_workflow_stage)
```

Comment:
[nitpick] 💅 

I think adding the column isn't necessary.

```suggestion
        label_hashes = filtered.index.str.split("_", n=1).str[0]
        lr_metadata = json.loads(pfs.label_row_meta.read_text(encoding="utf-8"))
        lr_to_workflow_stage = {
            lr_hash: metadata.get("workflow_graph_node", dict()).get("title", None)
            for lr_hash, metadata in lr_metadata.items()
        }
        filtered["workflow_stage"] = label_hashes.map(lr_to_workflow_stage)
```